packages/story-editor/src/components/panels/shared/captionsPanelContent.js

```diff
@@ -76,11 +76,15 @@ function CaptionsPanelContent({
   clearFileText = __('Remove file', 'web-stories'),
 }) {
   const {
+    allowedMimeTypes: { caption: allowedTextMimeTypes },
     capabilities: { hasUploadMediaAction },
     MediaUpload,
   } = useConfig();
 
-  if (!hasUploadMediaAction && !tracks.length) {
+  if (
+    (!hasUploadMediaAction && !tracks.length) ||
+    !allowedTextMimeTypes.length
```

Comment:
This hides the panel if `allowedTextMimeTypes` is empty but there are tracks.
If there are some tracks we should always display this panel.

So more `(!tracks.length && (!hasUploadMediaAction || !allowedTextMimeTypes.length))`

Reply:
This if state is to allow those with upload permission and have tracks set to remove caption files. The other statement is so that caption files are not be to uploaded to hide the upload button. This logic would not do this.